### PR TITLE
Refactor render args into a struct

### PR DESCRIPTION
**Pull request by jaderhs**
## Description

This change is a small improvement to change render args into a struct
This change will make it easier to add new render parameter without the need to update function calls in the tests that will not use the new parameter

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

**Review by tmjd — COMMENTED** [review]
Please add any updates as new commits so it will be easier to review.

File: pkg/controller/installation/core_controller.go [hunk 1]
```diff
@@ -1095,7 +1099,11 @@ func (r *ReconcileInstallation) Reconcile(ctx context.Context, request reconcile
 	// If we're on OpenShift on AWS render a Job (and needed resources) to
 	// setup the security groups we need for IPIP, BGP, and Typha communication.
 	if openShiftOnAws {
-		awsSetup, err := render.AWSSecurityGroupSetup(instance.Spec.ImagePullSecrets, &instance.Spec)
+		awsSGSetupCfg := &render.AWSSGSetupConfiguration{
+			PullSecrets: instance.Spec.ImagePullSecrets,
+			Installcr:   &instance.Spec,
```
[the remark this answers is not in the document]
Comment (jaderhs) [reply pkg/controller/installation/core_controller.go]: Renamed field Installcr to Installation

File: pkg/render/aws-securitygroup-setup.go [hunk 2]
```diff
@@ -28,24 +28,29 @@ import (
...
-func AWSSecurityGroupSetup(ps []corev1.LocalObjectReference, installcr *operatorv1.InstallationSpec) (Component, error) {
-	return &awsSGSetupComponent{pullSecrets: ps, installcr: installcr}, nil
+func AWSSecurityGroupSetup(cfg *AWSSGSetupConfiguration) (Component, error) {
+	return &awsSGSetupComponent{cfg: cfg}, nil
+}
+
+// AWSSGSetupConfiguration contains all the config information needed to render the component.
+type AWSSGSetupConfiguration struct {
+	PullSecrets []corev1.LocalObjectReference
+	Installcr   *operatorv1.InstallationSpec
```
[the remark this answers is not in the document]
Comment (jaderhs) [reply pkg/render/aws-securitygroup-setup.go]: Renamed field Installcr to Installation

File: pkg/render/fluentd.go [hunk 3]
```diff
@@ -145,29 +123,34 @@ type EksCloudwatchLogConfig struct {
 	FetchInterval int32
 }
 
+// FluentdConfiguration contains all the config information needed to render the component.
+type FluentdConfiguration struct {
+	LC              *operatorv1.LogCollector
```
[the remark this answers is not in the document]
Comment (jaderhs) [reply pkg/render/fluentd.go]: Renamed field `LC` to `LogCollector`

File: pkg/render/intrusion_detection.go [hunk 4]
```diff
@@ -49,60 +49,41 @@ const (
...
-		managedCluster:           managedCluster,
-		hasNoLicense:             hasNoLicense,
-		managerInternalTLSSecret: managerInternalTLSSecret,
+		cfg: cfg,
 	}
 }
 
+// IntrusionDetectionConfiguration contains all the config information needed to render the component.
+type IntrusionDetectionConfiguration struct {
+	LC                       *operatorv1.LogCollector
```
[the remark this answers is not in the document]
Comment (jaderhs) [reply pkg/render/intrusion_detection.go]: Renamed field `LC` to `LogCollector`

File: pkg/controller/logcollector/logcollector_controller.go [hunk 5]
```diff
@@ -419,20 +419,21 @@ func (r *ReconcileLogCollector) Reconcile(ctx context.Context, request reconcile
 	// Create a component handler to manage the rendered component.
 	handler := utils.NewComponentHandler(log, r.client, r.scheme, instance)
 
+	fluentdCfg := &render.FluentdConfiguration{
+		LC:              instance,
```
[the remark this answers is not in the document]
Comment (jaderhs) [reply pkg/controller/logcollector/logcollector_controller.go]: Renamed field `LC` to `LogCollector`

File: pkg/controller/intrusiondetection/intrusiondetection_controller.go [hunk 6]
```diff
@@ -360,20 +360,21 @@ func (r *ReconcileIntrusionDetection) Reconcile(ctx context.Context, request rec
...
-		pullSecrets,
-		r.provider == operatorv1.ProviderOpenShift,
-		r.clusterDomain,
-		esLicenseType,
-		managementClusterConnection != nil,
-		hasNoLicense,
-		managerInternalTLSSecret,
-	)
+	intrusionDetectionCfg := &render.IntrusionDetectionConfiguration{
+		LC:                       lc,
```
[the remark this answers is not in the document]
Comment (jaderhs) [reply pkg/controller/intrusiondetection/intrusiondetection_controller.go]: Renamed field `LC` to `LogCollector`

File: pkg/render/dex_test.go [hunk 7]
```diff
@@ -146,21 +153,22 @@ var _ = Describe("dex rendering tests", func() {
 				Effect:   corev1.TaintEffectNoExecute,
 			}
 
-			dexCfg := render.NewDexConfig(installation.CertificateManagement, authentication, tlsSecret, dexSecret, idpSecret, clusterName)
-			component := render.Dex(pullSecrets, false, &operatorv1.InstallationSpec{
-				ControlPlaneReplicas:    installation.ControlPlaneReplicas,
+			cfg.Installation = &operatorv1.InstallationSpec{
+				ControlPlaneReplicas:    cfg.Installation.ControlPlaneReplicas,
```
[the remark this answers is not in the document]
Comment (jaderhs) [reply pkg/render/dex_test.go]: Line removed.
The toleration is not assigned to `cfg.Installation.ControlPlaneTolerations`
Removed unused `Registry` field value

**Review by tmjd — APPROVED** [review]

**Pull request merged**